### ISSUE TITLE
trainning speed is slow, gpu occupancy is low

### ISSUE THREAD

**Issue by sunbaby01** [open-mmlab/mmpose#2342]
### Prerequisite

- [X] I have searched [Issues](https://github.com/open-mmlab/mmpose/issues) and [Discussions](https://github.com/open-mmlab/mmpose/discussions) but cannot get the expected help.
- [X] The bug has not been fixed in the latest version(https://github.com/open-mmlab/mmpose).

### Environment

OrderedDict([('sys.platform', 'linux'), ('Python', '3.9.12 (main, Jun  1 2022, 11:38:51) [GCC 7.5.0]'), ('CUDA av ailable', True), ('numpy_random_seed', 2147483648), ('GPU 0,1,2,3,4,5,6,7', 'NVIDIA A100-PCIE-40GB'), ('CUDA_HOME ', '/usr/local/cuda'), ('NVCC', 'Cuda compilation tools, release 11.4, V11.4.48'), ('GCC', 'gcc (Ubuntu 7.5.0-3ub untu1~18.04) 7.5.0'), ('PyTorch', '1.12.0'), ('PyTorch compiling details', 'PyTorch built with:\n  - GCC 9.3\n  - C++ Version: 201402\n  - Intel(R) oneAPI Math Kernel Library Version 2021.4-Product Build 20210904 for Intel(R) 64 architecture applications\n  - Intel(R) MKL-DNN v2.6.0 (Git Hash 52b5f107dd9cf10910aaa19cb47f3abf9b349815)\n - OpenMP 201511 (a.k.a. OpenMP 4.5)\n  - LAPACK is enabled (usually provided by MKL)\n  - NNPACK is enabled\n  - CPU capability usage: AVX2\n  - CUDA Runtime 11.3\n  - NVCC architecture flags: -gencode;arch=compute_37,code=sm_ 37;-gencode;arch=compute_50,code=sm_50;-gencode;arch=compute_60,code=sm_60;-gencode;arch=compute_61,code=sm_61;-g encode;arch=compute_70,code=sm_70;-gencode;arch=compute_75,code=sm_75;-gencode;arch=compute_80,code=sm_80;-gencod e;arch=compute_86,code=sm_86;-gencode;arch=compute_37,code=compute_37\n  - CuDNN 8.3.2  (built against CUDA 11.5) \n  - Magma 2.5.2\n  - Build settings: BLAS_INFO=mkl, BUILD_TYPE=Release, CUDA_VERSION=11.3, CUDNN_VERSION=8.3.2, CXX_COMPILER=/opt/rh/devtoolset-9/root/usr/bin/c++, CXX_FLAGS= -Wno-deprecated -fvisibility-inlines-hidden -DUSE _PTHREADPOOL -fopenmp -DNDEBUG -DUSE_KINETO -DUSE_FBGEMM -DUSE_QNNPACK -DUSE_PYTORCH_QNNPACK -DUSE_XNNPACK -DSYMB OLICATE_MOBILE_DEBUG_HANDLE -DEDGE_PROFILER_USE_KINETO -O2 -fPIC -Wno-narrowing -Wall -Wextra -Werror=return-type -Wno-missing-field-initializers -Wno-type-limits -Wno-array-bounds -Wno-unknown-pragmas -Wno-unused-parameter -W no-unused-function -Wno-unused-result -Wno-unused-local-typedefs -Wno-strict-overflow -Wno-strict-aliasing -Wno-e rror=deprecated-declarations -Wno-stringop-overflow -Wno-psabi -Wno-error=pedantic -Wno-error=redundant-decls -Wn o-error=old-style-cast -fdiagnostics-color=always -faligned-new -Wno-unused-but-set-variable -Wno-maybe-uninitial ized -fno-math-errno -fno-trapping-math -Werror=format -Werror=cast-function-type -Wno-stringop-overflow, LAPACK_ INFO=mkl, PERF_WITH_AVX=1, PERF_WITH_AVX2=1, PERF_WITH_AVX512=1, TORCH_VERSION=1.12.0, USE_CUDA=ON, USE_CUDNN=ON, USE_EXCEPTION_PTR=1, USE_GFLAGS=OFF, USE_GLOG=OFF, USE_MKL=ON, USE_MKLDNN=OFF, USE_MPI=OFF, USE_NCCL=ON, USE_NNP ACK=ON, USE_OPENMP=ON, USE_ROCM=OFF, \n'), ('TorchVision', '0.13.0'), ('OpenCV', '4.6.0'), ('MMEngine', '0.7.2'), ('MMPose', '1.0.0+ca37de9')])

mmcv                          2.0.0                                        
mmdet                         3.0.0                                        
mmengine                      0.7.2                                        
mmpose                        1.0.0                 
mmpretrain                    1.0.0rc7                                     

### Reproduces the problem - code sample

CUDA_VISIBLE_DEVICES=1 python tools/train.py td-hm_res50_8xb64-210e_coco-256x192.py --work-dir ./

### Reproduces the problem - command or script

CUDA_VISIBLE_DEVICES=1 python tools/train.py td-hm_res50_8xb64-210e_coco-256x192.py --work-dir .

### Reproduces the problem - error message

the traning speed is slow 
05/09 06:09:12 - mmengine - INFO - Epoch(train)   [2][ 700/2341]  lr: 5.000000e-04  eta: 2 days, 3:59:29  time: 0.383145  data_time: 0.179220  memory: 6560  loss: 0.001216  loss_kpt: 0.001216  acc_pose: 0.589468               

and gpu occupancy is 0- 50% ， it can't be higher.
I tried to add num_worker from 2 to 8, but it doesn't work.

**Comment by Tau-J** [comment]
Thanks for using MMPose. You can try larger batch size to increase the gpu usage ratio. In addition, training on COCO with a single GPU is supposed to be relatively slow, which seems to be normal.

**Comment by sunbaby01** [comment]
> Thanks for using MMPose. You can try larger batch size to increase the gpu usage ratio. In addition, training on COCO with a single GPU is supposed to be relatively slow, which seems to be normal.

thanks your reply. I double my bs from 64 to 128. It is a little improvement of gpu usage from50 to 65, but it is still always going to be 0, jumping between 60 and 0, and 0 is going to be a pretty big part of it. what can i do.

**Comment by sunbaby01** [comment]
another question, when in val phase , it use over 100 kernels of cpu. i set num_worker is 8.

**Comment by Tau-J** [comment]
> > Thanks for using MMPose. You can try larger batch size to increase the gpu usage ratio. In addition, training on COCO with a single GPU is supposed to be relatively slow, which seems to be normal.
> 
> thanks your reply. I double my bs from 64 to 128. It is a little improvement of gpu usage from50 to 65, but it is still always going to be 0, jumping between 60 and 0, and 0 is going to be a pretty big part of it. what can i do.

This phenomenon indicates that the bottleneck of your training is not on GPU (or say model inference), but CPU process like data augmentations, io, etc. which is not an error/bug of our repo. 
For the seconed question, you can raise an issue to mmengine.

**Comment by AnxQ** [comment]
I have countered the same problem. I migrated my code to the mmpose 1.0, the training speed is almost 5x slower. Both train time and data time are increased.

- MMPose 0.29 + MMCV-full 1.6.2
`2023-05-18 12:57:41,836 - mmpose - INFO - Epoch [1][100/586]	lr: 2.361e-06, eta: 19:48:10, time: 0.580, data_time: 0.034, memory: 10886, reg_loss: 4.9561, acc_pose: 0.0249, loss: 4.9561
2023-05-18 12:58:36,246 - mmpose - INFO - Epoch [1][200/586]	lr: 4.735e-06, eta: 19:10:39, time: 0.544, data_time: 0.000, memory: 10886, reg_loss: 4.7726, acc_pose: 0.0550, loss: 4.7726
2023-05-18 12:59:30,890 - mmpose - INFO - Epoch [1][300/586]	lr: 7.108e-06, eta: 18:59:04, time: 0.546, data_time: 0.000, memory: 10886, reg_loss: 4.6669, acc_pose: 0.0695, loss: 4.6669
20`

- MMPose 1.0 + MMEngine 0.7.3 + MMCV 2.0
`2023/05/19 09:36:24 - mmengine - INFO - Epoch(train)   [1][ 50/586]  lr: 1.177151e-06  eta: 2 days, 8:54:40  time: 1.665563  data_time: 0.969623  memory: 10387  grad_norm: 0.036514  loss: 0.465537  loss_kpt: 0.465537  acc_pose: 0.016770
2023/05/19 09:37:44 - mmengine - INFO - Epoch(train)   [1][100/586]  lr: 2.366204e-06  eta: 2 days, 7:55:55  time: 1.609580  data_time: 0.726436  memory: 10386  grad_norm: 0.021667  loss: 0.441042  loss_kpt: 0.441042  acc_pose: 0.029437
2023/05/19 09:39:04 - mmengine - INFO - Epoch(train)   [1][150/586]  lr: 3.555258e-06  eta: 2 days, 7:23:38  time: 1.592278  data_time: 0.535611  memory: 10386  grad_norm: 0.049266  loss: 0.427618  loss_kpt: 0.427618  acc_pose: 0.046950
`
(Calculation of the loss is a bit different but should not cause such a big difference)
The GPU utilization is volatile. The CPU 0 and CPU 1 utilization is 100% all the time, while the other cores are less than 30% on average.
I also noticed someone has the same problem in MMDetection: [#10329](https://github.com/open-mmlab/mmdetection/issues/10329). So I think there's something wrong with the MMEngine in specific cases.
